java/rocksjni/iterator.cc

```diff
@@ -121,6 +123,33 @@ void Java_org_rocksdb_RocksIterator_seek0(JNIEnv* env, jobject /*jobj*/,
   env->ReleaseByteArrayElements(jtarget, target, JNI_ABORT);
 }
 
+/*
+ * This method supports fetching into indirect byte buffers;
+ * the Java wrapper extracts the byte[] and passes it here.
+ *
+ * Class:     org_rocksdb_RocksIterator
+ * Method:    seek0
+ * Signature: (J[BII)V
+ */
+void Java_org_rocksdb_RocksIterator_seekByteArray0(
+    JNIEnv* env, jobject /*jobj*/, jlong handle, jbyteArray jtarget,
+    jint jtarget_off, jint jtarget_len) {
+  const std::unique_ptr<char[]> target(new char[jtarget_len]);
```

Comment:
I want to understand if this is correct.  Memory is being allocated here and freed at the end of this method.  But it is also being passed to GetByteArrayRegion.  Some comments here would help those (like me) who are less familiar with the JNIEnv understand what is going on.

Reply:
GetByteArrayRegion is a copy from the Java object into a C++ buffer (in this case the char[] contents). These are  being used as the slice key to the seek. I know you need to look up what GetByteArrayRegion does to grok this, but I think an accompanying comment would turn into an overly verbose half-wrong explanation of GetByteArrayRegion and I think the right thing is to let the code speak for itself, at the cost of a bit of reading time. I'm willing to be argued out of this position.